### DISCUSSION TITLE
GradientBoostingSurvivalAnalysis and SHAP

### DISCUSSION THREAD

**Discussion by Noomkwah** [sebp/scikit-survival#399] [Q&A]
Hello,

I am new to survival analysis, and I had the opportunity to use GradientBoostingSurvivalAnalysis in one of my projects.
I start this discussion because I would like to use SHAP package to explain the outputs of my model. As GradientBoostingSurvivalAnalysis  is currently not compatible with SHAP, I want to try to make it compatible. To do so, I have to clearly know how GradientBoostingSurvivalAnalysis  combines the outputs of its base learners.

The documentation states that "the predict_cumulative_hazard_function  calculates H(t|x) = exp( f(x) ) * H_0(t) where f(x) is the additive ensemble of base learners and H_0(t) is the baseline hazard function."

I tried to look into the code, but I could not find by myself. Let's introduce:
`gbm = GradientBoostingSurvivalAnalysis(n_estimators=5, learning_rate=0.1).fit(X_train, y_train)`

Does f(x) exactly equal the sum of the base learners predictions? For instance, would the following code work?
`x = X_test.iloc[0:1] # one data point`
`print('f(x) = ', sum([base_learner[0].predict(x) for base_learner in gbm.estimators_]))`

H_0(t) is estimated by the Breslow estimator and is not dependent on x (from my understanding). 

If I understood well, would one be able to explain the model `gbm` by explaining f(x) using TreeExplainer from SHAP?

**Comment by sebp** [comment]
The final prediction is a sum of base learners. But since it is gradient 
boosting, each base learner's prediction is scaled by the learning rate, 
see https://en.wikipedia.org/wiki/Gradient_boosting#Algorithm

In general, using SHAP is possible and users done it in the past:
https://github.com/sebp/scikit-survival/discussions/213

Whether TreeExplainer is suitable or not, I do not know.
